File: Extra Credit/JavaScript/app/entity/GameDifficulty.js
class _GameDifficulty {
    constructor(name) {
        this.name = name;
        Object.freeze(this);
    }
}

const GameDifficulty = Object.freeze({
    B: new _GameDifficulty("Beginner"),
    E: new _GameDifficulty("Easy"),
    N: new _GameDifficulty("Normal"),
    H: new _GameDifficulty("Hard"),
    I: new _GameDifficulty("Impossible"),
});

File: Extra Credit/Python/entity/gamedifficulty.py
import enum


class GameDifficulty(enum.Enum):
    B = "Beginner"
    E = "Easy"
    N = "Normal"
    H = "Hard"
    I = "Impossible"

    @classmethod
    def to_list(cls):
        difficulties = []
        for difficulty in cls:
            difficulties.append(difficulty)
        return difficulties

    def __str__(self):
        return self.value
